test/Solver/range1.ll

```diff
@@ -0,0 +1,23 @@
+; REQUIRES: solver
+
+; RUN: %llvm-as -o %t %s
+; RUN: %souper %solver -check %t 2> %t2 || true
+; RUN: %FileCheck %s < %t2
+
+define i1 @cmp_with_range(i8*, i8*) {
+  %v1 = load i8, i8* %0, !range !0
+  %v2 = load i8, i8* %1, !range !1
+  ; CHECK: instruction:
+  ; CHECK-NEXT: %out = icmp eq i8 %v1, %v2
+  ; CHECK-NEXT: unexpected simplification:

```

Comment:
also i'm not sure what it means to be checking for "unexpected simplification" -- why check for this? does it mean that you are checking to make sure that souper found an optimization that it was not supposed to find?